### PR TITLE
Setting up react router dom

### PR DESCRIPTION
**Pull request by hozayves**
## Purpose

The purpose of integrating React Router DOM is to enhance navigation within a React application, enabling smoother transitions between different pages or views.

## Changes Made

- Adjustments made to the Vite configuration.
- Modifications made in the `App.tsx` file to incorporate routing functionality.
- Creation of login pages to facilitate testing and authentication.

## Testing Instructions

To test the application:

1. Run the command `npm run dev`.
2. Navigate to the URL  `http://localhost:5001/login`, which should redirect you to the login page.

## Related Issues

List any related issues or tasks that were addressed during this implementation.

## Checklist

Before submitting, ensure the following:

- [ ] The code adheres to the project's coding standards.
- [ ] Changes are accompanied by relevant tests to maintain code integrity.
- [ ] Documentation has been updated to reflect the modifications made, if applicable.
- [ ] All tests pass to verify the stability and functionality of the application.

### CODE REVIEW

**Review by P-Rwirangira — APPROVED** [review]
Great work

**Review by JeanIrad — APPROVED** [review]

**Review by patrickhag — APPROVED** [review]
LGTM

File: vite.config.ts [hunk 1]
```diff
@@ -4,4 +4,7 @@ import react from "@vitejs/plugin-react-swc";
 // https://vitejs.dev/config/
 export default defineConfig({
   plugins: [react()],
+  server: {
+    port: 5001
```
Comment (Favor-star) [inline vite.config.ts:8]: why did you change the default port?

File: src/App.tsx [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+  const router = createBrowserRouter([
+    {
+      path: "/",
+      element: <SamplePage />
+    },
+    {
+      path: "login",
+      element: <Login />
+    }
+  ])
```
Comment (Favor-star) [inline src/App.tsx:15]: So by this kind of implementation, it means that whenever there are new routes, we will have to add them here? What if you create a separate file, and then export an array of routes that will be imported here, so that whenever there are new routes created  they would be added in that array, instead of being added here?
I personnaly think that this way should be more convenient, than the  one implemented.

Comment (hozayves) [reply src/App.tsx:15]: Yes Favour,
Thank you for your input. As our frontend grows larger, I anticipate that we will undertake a refactor.
